KubeArmor/core/kubeUpdate.go

```diff
@@ -1283,7 +1283,12 @@ func (dm *KubeArmorDaemon) ParseAndUpdateContainerSecurityPolicy(event tp.K8sKub
 	containername := ""
 	for k, v := range secPolicy.Spec.Selector.MatchLabels {
 		secPolicy.Spec.Selector.Identities = append(secPolicy.Spec.Selector.Identities, k+"="+v)
-		containername = v // TODO: fix identity behavior
+		if k == "kubearmor.io/container.name" {
+			containername = v
+		} else {
+			dm.Logger.Warnf("Fail to apply policy. The MatchLabels container name key should be `kubearmor.io/container.name` ")
+			return
```

Comment:
In future we might need to support other labels too like https://docs.docker.com/engine/reference/commandline/run/#set-metadata-on-container--l---label---label-file

So should we Warn here or just simply ignore?

Reply:
It would be a bit confusing for a user to actually know the reason for the failure of policy application. I think it's fine for now, we can update it in the future.